json/src/main/scala/io/sphere/json/generic/package.fmpp.scala

```diff
@@ -539,7 +539,10 @@ package object generic extends Logging {
   private def getJSONFields(clazz: Class[_]): IndexedSeq[JSONFieldMeta] = {
     Reflect.getCaseClassMeta(clazz).fields.map { fm =>
       val m = clazz.getDeclaredMethod(fm.name)
-      val name = Option(m.getAnnotation(classOf[JSONKey])).map(_.value).getOrElse(fm.name)
+      val name = Option(m.getAnnotation(classOf[JSONKey])) match {
```

Comment:
We could also remove patter matching and just check for null, though I'm not sure by how much it will improve the benchmark.

Reply:
I will give it a try, thanks for the idea.

I don't trust those benchmarks anymore, nothing seems to improve the scores 😄